sauce/features/budget/enter-to-move/index.js

```diff
@@ -0,0 +1,49 @@
+import { Feature } from 'toolkit/core/feature';
+import * as toolkitHelper from 'toolkit/helpers/toolkit';
+
+const MOVE_POPUP =
+  'ynab-u modal-popup modal-budget modal-budget-move-money ember-view modal-overlay active';
+const CATEGORY_DROPDOWN = 'dropdown-container categories-dropdown-container';
+const BUTTON_PRIMARY = 'button button-primary ';
+
+export class EnterToMove extends Feature {
+  modalIsOpen = false;
+
+  constructor() {
+    super();
+    this.onKeyDownHandler = this.onKeyDown.bind(this);
+  }
+
+  shouldInvoke() {
+    return toolkitHelper.getCurrentRouteName().indexOf('budget') !== -1;
+  }
+
+  invoke() {}
+
+  onKeyDown(e) {
+    const keycode = e.keycode || e.which;
+    if (keycode === 13) {
+      const OK = $(
+        '.modal-budget-move-money .modal-actions button:first-child'
+      );
+      OK.click();
+      document.removeEventListener('keydown', this.onKeyDownHandler, false);
+    }
+  }
+
+  observe(changedNodes) {
+    if (!this.shouldInvoke()) return;
+
+    if (changedNodes.has(MOVE_POPUP)) {
```

Comment:
Seems like `observe(changedNodes)` should look something like this or changedNodes would have to be passed to `invoke()`;
```
observe(changedNodes) {
  if (!this.shouldInvoke()) return;

  if (changedNodes.has(MOVE_POPUP) &&
      changedNodes.has(CATEGORY_DROPDOWN) &&
      changedNodes.has(BUTTON_PRIMARY)) {
    this.modalIsOpen = true;

    this.invoke();
  }
```
and `invoke()` would have this;
```
if (this.modalIsOpen) {
  document.addEventListener('keydown', this.onKeyDown, false);
  this.modalIsOpen = false;
}
```

Reply:
I originally had it in `invoke`, but it kept getting called when the page was ready, regardless of the logic I had in `observe`. This doesn't help me, because I only want the plugin to do something when the modal is open. So even though I want all my heavy lifting to be in `invoke`, right now it works best in `observe`. Any thoughts or solutions would be appreciated!

Reply:
`invoke()` should only be automatically called once during startup. After that it can be called again but it's typically done out of `observe()` when certain nodes have been changed or from `onRouteChanged()`. Many features have a check in `invoke()` to see if they've been through once already. A common check is to see if a feature specific element exists in the DOM and if not they add it. Subsequent checks detect it and skip code accordingly.

Reply:
That makes more sense. However - the way this code is structured, and the fact the only heavy lifting in `observe` is adding and removing an event listener, it feels like moving it into `invoke`, needing to pass `changedNodes`, and adding a DOM check is more complex than leaving it in `observe`.

Reply:
I wouldn't pass changedNodes to `invoke()`. I'd do it as I previously suggested which BTW I've updated to reflect your comment about the need for checking `this.modalIsOpen()` separately. You don't need a DOM check for this feature because you're not adding a visual element to the DOM.